test/unit/sample_type_test.rb

```diff
@@ -911,6 +911,61 @@ def setup
     assert_equal updated_at, sample.updated_at
   end
 
+  test 'adding a creator' do
+    User.current_user = @person.user
+    creator = Factory :person
+    sample_type = Factory(:simple_sample_type, contributor: @person)
+    params = { creator_ids: [creator.id] }
+    assert_difference('sample_type.creators.count') do
+      assert_difference('AssetsCreator.count') do
+        sample_type.update!(params)
+      end
+    end
+    sample_type.save!
```

Comment:
needs an assertion that the creators have been set correctly

```
sample_type.reload
assert_equal [creator], sample_type.creators
```
`.reload` resets the model, and reflects what is in the DB rather than cached in memory